AAI/README.md

```diff
@@ -95,6 +90,21 @@ proxies for "Claim Consumption Only" functionality --
 Data holders and data owners should explore their options to decide what best
 fits their needs.
 
+#### Embedded Tokens example and explanation
+
+![embedded](https://github.com/ga4gh/data-security/blob/master/AAI/embedded_Claims_flow.png)
+
+Consider two parties: Google and ELIXIR.  
+
+In this example, Google Passport Clearinghouse makes access decisions based on ELIXIR Assertion Repository information via a chain of brokers that have passed along the Passport Visas in standard GA4GH Passport format where the Passports are signed by different Brokers but the Passport Visas retain the signature from the Passport Visa Issuer. 
+
+The way this chain of brokers and trust is maintained is through "embedded tokens". There are two types of embedded tokens: Embedded Access Tokens and Embedded Document Tokens. 
+
+Embedded Access Tokens are claims in a Broker's token that can then be sent to OTHER brokers' `/userinfo` endpoints for further user claims. 
```

Comment:
This sounds a little like the former embedded token scheme that we have now replaced. Perhaps some new wording could distinguish what is provided now more clearly:

"Embedded Access Tokens allow a Visa Issuer the capability to revoke the claim or update the contents of the claim periodically by using OIDC discovery on the issuer and then calling /userinfo with the access token and examining the results. It is not a full passport, so the /userinfo endpoint will only return the set of GA4GH claims that were in the original Embedded Access Token, or return a non-retryable error to indicate the token is no longer valid."

Reply:
BTW, Embedded Access Tokens now also have all the claims in them already and do not require any calls to /userinfo except to verify validity and update the content.

Reply:
Also note that not all libraries will be able to update the content and this is not a requirement in the spec for embedded access token. Only that the /userinfo returns valid or invalid.

Reply:
> "Embedded Access Tokens allow a Visa Issuer the capability to revoke the claim or update the contents of the claim periodically by using OIDC discovery on the issuer 

?? Visa issuers (in contrast to Passport issuers) are not running OP so there is no OIDC discovery

>will only return the set of GA4GH claims that were in the original Embedded Access Token, 

?? (embedded) access tokens do not have GA4GH claims

Reply:
@cdvoisin I agree with @mikael-linden here -- what's your understanding? I was conveying the difference between Access and DOcument tokens. Access tokens are just that -- access tokens and do not have all the claims.

Reply:
You are right, the AAI spec makes those claims optional. However the RI spec makes them required as per: https://github.com/ga4gh-duri/ga4gh-duri.github.io/blob/master/researcher_ids/ga4gh_passport_v1.md#passport-visa-format

This avoids the case where a  clearinghouse needs to fetch /userinfo endpoints for each of a 1000 visas to determine if you have access or not. These are small enough to just embed them in the token. /userinfo is used for revocation (instead of /introspect) as well as some refreshing of content capabilities (https://github.com/ga4gh/data-security/blob/master/AAI/AAIConnectProfile.md#at-polling item "v").

Furthermore, if you are using Embedded Access Tokens, you need OIDC Discovery to find the /userinfo endpoint. The point of Embedded Document Tokens is that you don't have these extra requirements and only need to worry about public key distribution via the "jku" (or assume an out-of-band distribution).

Let me know if this is not your read of the spec, or if you have concerns about this approach.

Reply:
Furthermore, if the implementer's library or service does not support adding the claim in the Embedded Access Token as well as refreshing it via /userinfo, the Embedded Document Token format is still a viable alternative.

Reply:
Ok. Just pushed what I think is a good fix. I'm resolving this conversation and please reopen if you don't find it satisfactory.